sympy/physics/vector/dyadic.py

```diff
@@ -539,6 +539,50 @@ def _eval_evalf(self, prec):
             new_args.append(tuple(new_inlist))
         return Dyadic(new_args)
 
+    def xreplace(self, rule):
+        """
+        Replace occurrences of objects within the expression.
+
+        Parameters
+        ==========
+
+        rule : dict-like
+            Expresses a replacement rule
+
+        Returns
+        =======
+
+        xreplace : the result of the replacement
+
+        Examples
+        ========
+
+        >>> from sympy import symbols, pi
+        >>> from sympy.physics.vector import ReferenceFrame
+        >>> N = ReferenceFrame('N')
+        >>> x, y, z = symbols('x y z')
+        >>> ((1 + x*y) * (N.x | N.x)).xreplace({x: pi})
+        (pi*y + 1)*(N.x|N.x)
+        >>> ((1 + x*y) * (N.x | N.x)).xreplace({x: pi, y: 2})
+        (1 + 2*pi)*(N.x|N.x)
```

Comment:
Avoid the use of `|` for operations in the documentation. I plan to deprecate these overloaded operations eventually.